main.nf

```diff
@@ -0,0 +1,517 @@
+        --processes 1 --b_thr $params.b_thr --nbr_dir $params.nbr_dir \$ball_stick_arg \
+        --para_diff $params.para_diff \$perp_diff --iso_diff $params.iso_diff \
+        --save_kernels kernels/ --compute_only
+    """
+}
+
+data_for_commit
+    .join(h5_for_commit)
+    .combine(kernel_for_commit)
+    .set{data_tracking_kernel_for_commit}
+
+process Run_COMMIT {
+    cpus params.processes_commit
+
+    input:
+    set sid, file(bval), file(bvec), file(dwi), file(peaks), file(h5), file(kernels) from data_tracking_kernel_for_commit
+
+    output:
+    set sid, "${sid}__results_bzs/"
+    set sid, "${sid}__decompose_commit.h5" into h5_for_afd_rd
+
+    when:
+    params.run_commit
+
+    script:
+    """
+    ball_stick_arg=""
+    perp_diff=""
+    if $params.ball_stick; then
+        ball_stick_arg="--ball_stick"
+    else
+        perp_diff="--perp_diff $params.perp_diff"
+    fi
+    scil_run_commit.py $h5 $dwi $bval $bvec "${sid}__results_bzs/" --in_peaks $peaks \
+        --processes $params.processes_commit --b_thr $params.b_thr --nbr_dir $params.nbr_dir \$ball_stick_arg \
+        --para_diff $params.para_diff \$perp_diff --iso_diff $params.iso_diff --load_kernel $kernels
+    mv "${sid}__results_bzs/decompose_commit.h5" ./"${sid}__decompose_commit.h5"
+    """
+}
+
+if (!params.run_commit) {
+    h5_for_skip_commit
+        .into{h5_for_afd_rdl;h5_for_skip_aft_rd}
+}
+
+h5_for_afd_rd
+    .join(fodf_for_afd_rd)
+    .set{h5_fodf_for_afd_rd}
+
+process Compute_AFD_RD {
+    cpus params.processes_afd_rd
+
+    input:
+    set sid, file(h5), file(fodf) from h5_fodf_for_afd_rd
+
+    output:
+    set sid, "${sid}__decompose_afd_rd.h5" into h5_for_transformation
+
+    when:
+    params.run_afd_rd
+
+    script:
+    """
+    length_weighting_arg=""
+    if $params.length_weighting; then
+        length_weighting_arg="--length_weighting"
+    fi
+    scil_compute_mean_fixel_afd_from_hdf5.py $h5 $fodf "${sid}__decompose_afd_rd.h5" \$length_weighting_arg \
+        --sh_basis $params.sh_basis --processes $params.processes_afd_rd
+    """
+}
+
+anat_for_registration
+    .combine(template_for_registration)
+    .set{anats_for_registration}
+process Register_Anat {
+    cpus params.processes_register
+    input:
+    set sid, file(native_anat), file(template) from anats_for_registration
+
+    output:
+    set sid, "${sid}__output0GenericAffine.mat", "${sid}__output1Warp.nii.gz", "${sid}__output1InverseWarp.nii.gz"  into transformation_for_data, transformation_for_metrics
+    file "${sid}__outputWarped.nii.gz"
+    script:
+    """
+    antsRegistrationSyNQuick.sh -d 3 -m ${native_anat} -f ${template} -n ${params.processes_register} -o "${sid}__output" -t s
+    """ 
+}
+
+in_opt_metrics
+    .transpose()
+    .concat(anat_for_metrics)
+    .groupTuple()
+    .flatMap{ sid, metrics -> metrics.collect{ [sid, it] } }
+    .combine(transformation_for_metrics, by: 0)
```

Comment:
This is a join.

Reply:
Does not work, there is one transformation and multiple metrics